### PR TITLE
Add plain fasthttp

### LINKED ISSUE #1915: Add fasthttp
Hi @valyala,

Would you mind to add (or consent that we add) [fasthttp](https://github.com/valyala/fasthttp) in `here` ?

This project compare performance based on 3 routes :

+ `GET` `/`, **SHOULD** return a successful status code and an empty string
+ `GET` `/user/:id`, **SHOULD** return a successful status code and the id
+ `POST` `/user`, **SHOULD** return a successful status code and an empty string

Regards,

### PR DESCRIPTION
**Pull request by erikdubbelboer**
Improve fasthttprouter, fmt is slow.

Fixes #1915

### CODE REVIEW

File: go/fasthttp/go.mod [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module main
+
+require github.com/valyala/fasthttp v1.5.0
+
+go 1.13
```
Comment (waghanza) [inline go/fasthttp/go.mod]: this line is not required

File: go/fasthttp/go.sum [hunk 2]
```diff
@@ -0,0 +1,13 @@
+github.com/klauspost/compress v1.8.2 h1:Bx0qjetmNjdFXASH02NSAREKpiaDwkO1DRZ3dV2KCcs=
```
Comment (waghanza) [inline go/fasthttp/go.sum]: keeping the `.sum` is useful, but not for this use-case

File: go/fasthttprouter/go.mod [hunk 3]
```diff
@@ -1,4 +1,8 @@
 module main
 
+require (
+	github.com/buaazp/fasthttprouter v0.1.1
+	github.com/valyala/fasthttp v1.5.0
+)
 
-require github.com/buaazp/fasthttprouter v0.1.1
\ No newline at end of file
+go 1.13
```
Comment (waghanza) [inline go/fasthttprouter/go.mod]: not required

File: go/fasthttprouter/go.sum [hunk 4]
```diff
@@ -0,0 +1,15 @@
+github.com/buaazp/fasthttprouter v0.1.1 h1:4oAnN0C3xZjylvZJdP35cxfclyn4TYkW6Y+DSvS+h8Q=
```
Comment (waghanza) [inline go/fasthttprouter/go.sum]: keeping the `.sum` is useful, but not for this use-case

File: go/fasthttprouter/main.go [hunk 5]
```diff
@@ -1,22 +1,20 @@
...
-
 	"github.com/buaazp/fasthttprouter"
 	"github.com/valyala/fasthttp"
 )
 
 func main() {
 	router := fasthttprouter.New()
 	router.GET("/", func(ctx *fasthttp.RequestCtx) {
-		fmt.Fprint(ctx, "")
+		ctx.SetBodyString("")
```
Comment (waghanza) [inline go/fasthttprouter/main.go]: thanks :tada:

File: go/fasthttp/main.go [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+	"bytes"
+
+	"github.com/valyala/fasthttp"
+)
+
+var (
+	strGET            = []byte("GET")
+	strPOST           = []byte("POST")
+	strSlash          = []byte("/")
+	strSlashUserSlash = []byte("/user/")
```
Comment (waghanza) [inline go/fasthttp/main.go]: is there any way to have a dynamic URL ?

I mean `strSlashUserSlash` could lead either to `/user/0` than `/user/$hacking_value*` ?

File: go/fasthttp/go.mod (hunk 1 above)
Comment (erikdubbelboer) [reply go/fasthttp/go.mod]: It's automatically added by Go 1.13 each time you run any of the `go mod` commands.

File: go/fasthttprouter/go.sum (hunk 4 above)
Comment (erikdubbelboer) [reply go/fasthttprouter/go.sum]: The whole idea behind the `go.sum` file is to commit it so you have reproducible builds that always use the same dependencies.

File: go/fasthttp/main.go (hunk 6 above)
Comment (erikdubbelboer) [reply go/fasthttp/main.go]: That's why I'm using `bytes.HasPrefix` to check. There is no default router build into fasthttp.

File: go/fasthttprouter/go.sum (hunk 4 above)
Comment (waghanza) [reply go/fasthttprouter/go.sum]: sure, I know, like a `lock` file in `ruby` ..., but this also means that be versioned ... as this is not an end application, the benefit is not accurate `here`

could it be removed ?

PS : We have no `.sum` files here, I think it's **NOT** needed here :stuck_out_tongue:

File: go/fasthttp/main.go (hunk 6 above)
Comment (waghanza) [reply go/fasthttp/main.go]: ok, it seems different than other frameworks I know :heart:

**Comment by erikdubbelboer** [comment]
I have removed the things you requested.

**Pull request merged**